### PR TITLE
Remove redundant functionality from git-hack

### PR DESCRIPTION
**Pull request by kevgo**
Concludes #778 and #779 by removing the functionality now present in `git append` and `git prepend`. Overall a nice simplification!

@charlierudolph

### CODE REVIEW

File: src/git-town-hack [hunk 1]
```diff
@@ -15,17 +15,7 @@ function preconditions {
...
-    parent_branch_name=$MAIN_BRANCH_NAME
-  elif [ "$parent_branch_name" == '.' ]; then
-    parent_branch_name=$INITIAL_BRANCH_NAME
-  fi
-  ensure_has_branch "$parent_branch_name"
-  if [ "$parent_branch_name" != "$MAIN_BRANCH_NAME" ]; then
-    ensure_knows_parent_branches "$parent_branch_name"
-  fi
-
+  parent_branch_name=$MAIN_BRANCH_NAME
```
Comment (charlierudolph) [inline src/git-town-hack]: I think this variable can be removed and we can just use `$MAIN_BRANCH_NAME` in the steps.

Comment (kevgo) [inline src/git-town-hack]: Okay, done

**Comment by kevgo** [comment]
All comments addressed

**Comment by charlierudolph** [comment]
@kevgo was this shipped?

**Comment by kevgo** [comment]
Shipped now.

**Pull request closed without merging**